var/spack/repos/builtin/packages/precice/package.py

```diff
@@ -33,38 +33,68 @@ class Precice(CMakePackage):
     variant('python', default=False, description='Enable Python support')
     variant('shared', default=True, description='Build shared libraries')
 
-    # Not yet
-#    variant(
-#        'float', default=False,
-#        description='Use single precision for field data exchange')
-#    variant(
-#        'int64',
-#        default=False, description='Use 64-bit integers for indices')
-
     depends_on('cmake@3.5:', type='build')
-    depends_on('cmake@3.9.6:', type='build', when='@1.4:')
+    depends_on('cmake@3.10.2:', type='build', when='@1.4:')
     depends_on('boost@1.60.0:')
     depends_on('boost@1.65.1:', when='@1.4:')
     depends_on('eigen@3.2:')
+    depends_on('eigen@:3.3.7', type='build', when='@:1.5')  # bug in prettyprint
     depends_on('libxml2')
     depends_on('mpi', when='+mpi')
     depends_on('petsc@3.6:', when='+petsc')
-    depends_on('python@2.7', when='+python', type=('build', 'run'))
+    depends_on('python@2.7.16', when='+python', type=('build', 'run'))
+    depends_on('py-numpy@1.16.4', when='+python', type=('build', 'run'))
```

Comment:
It doesn't work with any other version of numpy?

Reply:
If I don't fix the numpy version, I get the following error:

<details>
<summary> click to expand </summary>

```
==> Error: An unsatisfiable version constraint has been detected for spec:

    python@2.7.16%gcc@9.1.0+bz2+ctypes+dbm+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-ubuntu16.04-x86_64
        ^bzip2
            ^diffutils
        ^expat
        ^gdbm
            ^readline
                ^ncurses
                    ^pkgconf@1.6.1%gcc@9.1.0 arch=linux-ubuntu16.04-x86_64
        ^gettext@0.19.8.1%gcc@9.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu16.04-x86_64
            ^libxml2@2.9.9%gcc@9.1.0~python arch=linux-ubuntu16.04-x86_64
                ^libiconv@1.15%gcc@9.1.0 arch=linux-ubuntu16.04-x86_64
                ^xz@5.2.4%gcc@9.1.0 arch=linux-ubuntu16.04-x86_64
                ^zlib@1.2.11%gcc@9.1.0+optimize+pic+shared arch=linux-ubuntu16.04-x86_64
            ^tar
        ^libffi
        ^openssl
            ^perl@5.14.0:
        ^sqlite@3.0.8:


while trying to concretize the partial spec:

    py-numpy@1.17.0%gcc@9.1.0+blas+lapack arch=linux-ubuntu16.04-x86_64

py-numpy requires python version 3.5:, but spec asked for 2.7.16
```
</details>

Reply:
In that case you want:
```suggestion
    # numpy 1.17+ requires Python 3
    depends_on('py-numpy@:1.16', when='+python', type=('build', 'run'))
```